checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -46,8 +47,20 @@ public UpperBoundTransfer(CFAnalysis analysis) {
             AssignmentNode node, TransferInput<CFValue, CFStore> in) {
         TransferResult<CFValue, CFStore> result = super.visitAssignment(node, in);
 
-        if (node.getExpression() instanceof ArrayCreationNode) {
-            ArrayCreationNode acNode = (ArrayCreationNode) node.getExpression();
+        Node expNode = node.getExpression();
+
+        ArrayCreationNode acNode =
+                expNode instanceof ArrayCreationNode ? (ArrayCreationNode) expNode : null;
+
+        if (expNode instanceof TypeCastNode) {
+            Node innerExpNode = ((TypeCastNode) expNode).getOperand();
+            acNode =
+                    innerExpNode instanceof ArrayCreationNode
+                            ? (ArrayCreationNode) innerExpNode
+                            : null;
```

Comment:
Under what circumstances can the result be `null`?  I would think you would get to an `ArrayCreationNode` eventually.  Please remove this possibility or document under what circumstances you cannot get to an `ArrayCreationNode`.